log/logger.py

```diff
@@ -189,15 +192,23 @@ def step_info(self, step_name):
     def get_additional_logs(self):
         from core.test_run import TestRun
         from test_tools.fs_tools import check_if_file_exists
+
         messages_log = "/var/log/messages"
         if not check_if_file_exists(messages_log):
             messages_log = "/var/log/syslog"
-        log_files = {"messages.log": messages_log,
+
+        log_files = {"messages.log": "/tmp/messages",
                      "dmesg.log": "/tmp/dmesg"}
         extra_logs = TestRun.config.get("extra_logs", {})
         log_files.update(extra_logs)
 
-        TestRun.executor.run(f"dmesg > {log_files['dmesg.log']}")
+        # Escape special characters from test identifier to be properly processed by awk
+        test_identifier = re.escape(TestRun.LOGGER.unique_test_identifier)
+
+        TestRun.executor.run(
+            f"dmesg | awk '/{test_identifier}/,0' > {log_files['dmesg.log']}")
+        TestRun.executor.run(
+            f"awk '/{test_identifier}/,0' {messages_log} > {log_files['messages.log']}")
```

Comment:
If the identifier is set to the template and no one calls print_test_identifier_to_logs() then there is no identifier in logs and awk returns empty string.
I'd suggest default empty value and if the identifier is empty gather everything - otherwise use this new way with awk